pages/api/users/[uuid].ts

```diff
@@ -57,17 +58,27 @@ async function handleGET({ res, queryUUID }: HandlerParams<User>) {
   return res.status(200).json(user);
 }
 
-async function handlePATCH({ req, res, queryUUID }: HandlerParams<User>) {
-  const newUserDetails = req.body as Partial<User>;
+async function handlePATCH({ req, res, userUUID }: HandlerParams<User>) {
+  // CHECK THIS
+  // tämä .pick-tapa vai kokonaan uuden scheman luonti?
+  const userData = userSchema
+    .pick({ email: true, firstName: true, lastName: true })
+    .safeParse(req.body);
+
+  if (!userData.success) {
+    throw new HttpError('Invalid request body!', 400);
+  }
+
+  const { email, firstName, lastName } = userData.data;
```

Comment:
Olisiko järkevämpi tehdä `handlePATCH`:lle ja `handlePUT`:lle omat schemat, eikä käyttää tuota `.pick`-tapaa?

Reply:
Parempi tehdä omat skeemat käyttäen`zod`in `extend` ja `merge` ominaisuuksia. Ks. `zodSchemas.ts` tiedoston kommentti missä määritetty `updateUserSchema`.